### PR TITLE
[111] Gradle scripts with Kotlin DSL - Part 3 (Jacoco and Checkstyle gradle)

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
https://github.com/nimblehq/android-templates/issues/111

## What happened 👀

Kotlin DSL is now supported to handle Gradle scripts with Kotlin syntax. It's more familiar than Groovy-based syntax for Android developers.

-> Convert some common Groovy-based Gradle scripts into Kotlin DSL scripts.
 
## Insight 📝

- Migrate `jacoco.gradle` & `codequality.gradle` to `.kts`

 
## Proof Of Work 📹

🚀 CI build should be passed ✅

### CODE REVIEW

File: RxJavaTemplate/config/jacoco.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        xml.isEnabled = true
+        html.isEnabled = true
+        csv.isEnabled = false
+    }
+}
+
+tasks.withType<Test> {
+    configure<JacocoTaskExtension> {
+        isIncludeNoLocationClasses = true
+        /**
```
Comment (lydiasama) [inline RxJavaTemplate/config/jacoco.gradle.kts:84]:
```suggestion
        /*
```
We use this comment style inside of the block, right? 😄

**Review by lydiasama — COMMENTED** [review]
The rest LGTM. 🚀

File: RxJavaTemplate/config/jacoco.gradle.kts (hunk 1 above)
Comment (hoangnguyen92dn) [reply RxJavaTemplate/config/jacoco.gradle.kts:84]: @lydiasama I migrated this comment from @manh-t 's implementation🥲

**Review by Tuubz — APPROVED** [review]
Rest LGTM 🤖

File: RxJavaTemplate/config/jacoco.gradle.kts (hunk 1 above)
Comment (luongvo) [reply RxJavaTemplate/config/jacoco.gradle.kts:84]: Time to fix that @hoangnguyen92dn 😆

File: RxJavaTemplate/config/jacoco.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ */
+val executionDataTree = fileTree(project.rootDir) {
+    include(
+        "app/jacoco.exec",
+        "common/jacoco.exec"
+    )
+}
+
+tasks.register<JacocoReport>("jacocoTestReport") {
+
```
Comment (luongvo) [inline RxJavaTemplate/config/jacoco.gradle.kts:61]: why do we need 1 linebreak here

**Review by luongvo — COMMENTED** [review]
rest lgtm

**Review by luongvo — APPROVED** [review]
lgtm

**Review by lydiasama — APPROVED** [review]

**Pull request merged**